patterns/factorymethod/shipyards/src/main/java/com/scottlogic/dp/factorymethod/shipyards/game/GameMapInterface.java

```diff
@@ -30,7 +30,12 @@ public Shipyard selectShipyard(String name) {
         }
     }
 
+    public void upgrade(Shipyard shipyard) {
+        shipyard.upgrade();
+        System.out.println("Upgrade Shipyard");
```

Comment:
Interesting, now we have some shipyards producing old ships and some of them new ships. Why not! (originally I thought that every shipyard gets upgraded, hence the signature of the method `upgrade()` :) )